Code/Robot/Robot/Robot.ino

```diff
@@ -126,3 +84,79 @@ void read_serial(JsonDocument& doc) {
     return;
   }
 }
+
+void driving_logic(queue<Move>* moveQueue) {
+  float motorSpeeds[NUMBER_OF_MOTORS]; // Initialize motor speeds
+
+  Move nextMove = getNextMoveFromQueue(moveQueue);
+  int time = nextMove.time;
+  switch (nextMove.direction) {
+    case 1:  // forwards
+      wheelbase->computeWheelSpeeds(10, 0, 0, motorSpeeds);
+      run_motors_with_blocking_delay(time, motorSpeeds, false);
+      break;
+    case 2:  // left
+      wheelbase->computeWheelSpeeds(0, -10, 0, motorSpeeds);
+      run_motors_with_blocking_delay(time, motorSpeeds, false);
+      break;
+    case 3:  // backwards
+      wheelbase->computeWheelSpeeds(-10, 0, 0, motorSpeeds);
+      run_motors_with_blocking_delay(time, motorSpeeds, false);
+      break;
+    case 4:  // right
+      wheelbase->computeWheelSpeeds(0, 10, 0, motorSpeeds);
+      run_motors_with_blocking_delay(time, motorSpeeds, false);
+      break;
+    case 5:  // lift motor
+      run_motors_with_blocking_delay(time, nullptr, true);
+      break;
+    case 6:  // belt motor
+      run_motors_with_blocking_delay(time, nullptr, false);
+      break;
+    default:
+      Serial.println("Unexpected input in direction switch");
+      Serial.println(nextMove.direction);
+      break;
+  }
+}
+
+Move getNextMoveFromQueue(queue<Move>* queueToPopFrom) {
+  Move retMove = queueToPopFrom->back();
+  queueToPopFrom->pop();
+  return retMove;
+}
+
+void run_motors_with_blocking_delay(int delayTime, float* motorSpeeds, bool lift_motor) {
```

Comment:
If the input to this was an int[6], we could set the corresponding motors (including lift motors) more cleanly. 

* with that said, I'm a big fan of this function